### PR TITLE
Fixes #7578

### LINKED ISSUE #7578: Making contributions should be easier
### Expected behavior
Someone intending to contribute should be able to set up their development environment quickly and easily. 

### Actual behavior
Development environment [setup](http://netty.io/wiki/setting-up-development-environment.html) requires several manual steps:
- Local Maven installation is required.
- Git line-ending manual configuration is required.

### PR DESCRIPTION
**Pull request by asarkar**
Motivation:
Someone intending to contribute should be able to set up their development environment quickly and easily.

Modifications:
- Added a Maven wrapper such that a local Maven installation isn't necessary.
- Added a .gitattributes such that auto line-endings are enforced.

Result:
Fixes #7578. 

- ./mvnw is enough to build.
- Git line-endings are enforced.

### CODE REVIEW

File: mvnw [hunk 1]
```diff
@@ -0,0 +1,227 @@
+#!/bin/sh
+# ----------------------------------------------------------------------------
```
Comment (normanmaurer) [inline mvnw]: @asarkar was this forked from somewhere ?

File: mvnw.cmd [hunk 2]
```diff
@@ -0,0 +1,145 @@
+@REM ----------------------------------------------------------------------------
+@REM Licensed to the Apache Software Foundation (ASF) under one
```
Comment (normanmaurer) [inline mvnw.cmd]: was this forked from somewhere ?

Comment (asarkar) [reply mvnw.cmd]: https://github.com/takari/maven-wrapper

File: mvnw (hunk 1 above)
Comment (asarkar) [reply mvnw]: https://github.com/takari/maven-wrapper

**Comment by asarkar** [comment]
I also considered putting the code style and inspection profile in the project, but that'd require committing `.idea`.

**Comment by asarkar** [comment]
This PR is ready to be merged. Do you've any more comments/questions?

**Comment by normanmaurer** [comment]
Honestly I am not sure if this is really needed as we have a docker image for people that not want to install maven by themselves. They will need to install multiple other things as well at the end to build the native transport.

@Scottmitch @trustin @carl-mastrangelo @ejona86 WDYT ?

**Comment by asarkar** [comment]
AFAIK, the Docker image had not been released to Docker Hub, so one still has to build it locally.

**Comment by normanmaurer** [comment]
Not sure what is the big deal of building it locally ?

**Comment by asarkar** [comment]
It’s not a problem, but it’s still a manual step. The idea workflow would be for someone to simply pull and start and start coding (and building). Zero local configuration necessary.

**Comment by ejona86** [comment]
The Gradle wrapper (which the Maven wrapper seems a clone of) has been really nice in gRPC. It means users no longer care what version of gradle we expect, which also helps the CI infrastructure. I'm a strong believer in it.

Doing development in Docker is okay, but it's something that's hard to force on others in my mind. To do it well, it requires Docker knowledge, binding paths, maybe multiple terminals, and forces me to use your Linux distribution. It's certainly nice to get _something_ running quickly, but it's not a panacea.

Part of the problem with Maven is getting an up-to-date version. Many distros will have old versions and there's no clear error messages when things are too old. You then get to manage your own version, which has the same problem of getting old.

File: mvnw (hunk 1 above)
Comment (normanmaurer) [reply mvnw]: @asarkar you will add it to the NOTICE.txt file and license folder tho

File: mvnw.cmd (hunk 2 above)
Comment (normanmaurer) [reply mvnw.cmd]: See above

**Comment by normanmaurer** [comment]
@asarkar please let me know once you addressed my comment related to NOTICE.txt etc.

**Comment by asarkar** [comment]
@normanmaurer Updated NOTICE and added a license.

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (f99a1985ce92ff6fb378aa1240defc1792dbd742) and 4.0 (141b674ad)

**Comment by normanmaurer** [comment]
@asarkar thanks

**Pull request closed without merging**